### PR TITLE
[ra2] update to k8s 1.24

### PR DESCRIPTION
**Pull request by rgstori**
main difference is that beta apis will be disabled by default starting k8s 1.24

### CODE REVIEW

File: doc/ref_arch/kubernetes/chapters/chapter06.rst [hunk 1]
```diff
@@ -23,13 +23,14 @@ run with different Feature Gate settings.
...
-  changes may happen)
+- Beta features are disabled by default, are well tested, and support will not be dropped (although breaking API
+  changes may happen). Any existing Beta feature as of 1.24 will continue to be enabled by default, but new beta APIs
+  and features will not be enabled by default.
 - GA features are stable, always enabled and cannot be disabled.
 
 The policy for RA2 to include Kubernetes features as mandatory is:
 
-   Only features that are either in Beta or GA stage can be made mandatory, subject to RA2 requirements.
+   Only features that are either in GA stage (or Beta before 1.24) can be made mandatory, subject to RA2 requirements.
```
Comment (rgstori) [inline doc/ref_arch/kubernetes/chapters/chapter06.rst:33]:
```suggestion
   Only features that are in GA stage (or Beta before 1.24) can be made mandatory, subject to RA2 requirements.
```

**Review by collivier — CHANGES_REQUESTED** [review]
They are missing changes. see node labels for instance.

File: doc/ref_arch/kubernetes/chapters/chapter06.rst [hunk 2]
```diff
@@ -23,13 +23,14 @@ run with different Feature Gate settings.
 A feature can be in Alpha, Beta or GA stage:
 
 - Alpha features are disabled by default, may be buggy, and support may be dropped
-- Beta features are enabled by default, are well tested, and support will not be dropped (although breaking API
-  changes may happen)
+- Beta features are disabled by default, are well tested, and support will not be dropped (although breaking API
+  changes may happen). Any existing Beta feature as of 1.24 will continue to be enabled by default, but new beta APIs
+  and features will not be enabled by default.
```
Comment (petorre) [inline doc/ref_arch/kubernetes/chapters/chapter06.rst:28]:
```suggestion
  and features will not be enabled by default. More in [KEP-3136](https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/3136-beta-apis-off-by-default/README.md).
```

**Comment by rgstori** [comment]
> They are missing changes. see node labels for instance.

@collivier - not sure I understand what you mean - there's node labels specs in RA2 for Anuket profiles and nothing else, so these have not changed in k8s 1.24.

if you're referring to [KEP-2067: Rename the kubeadm "master" label and taint](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/kubeadm/2067-rename-master-label-taint), there's no impact on RA2

**Review by petorre — APPROVED** [review]

**Review by collivier — CHANGES_REQUESTED** [review]
https://kubernetes.io/blog/2022/04/07/upcoming-changes-in-kubernetes-1-24/
https://github.com/kubernetes/kubernetes/pull/107533

```
chapters/chapter06.rst:     - ServiceAccount admission controller migration master upgrade should maintain a functioning cluster
chapters/chapter06.rst:     - Master upgrade should NOT disrupt GPU Pod
chapters/chapter07.rst:   This produces an overhead of Kubernetes Masters - is there a way of making this more efficient whilst still able to
chapters/chapter04.rst:     - A deb/rpm compatible distribution of Linux (this must be used for the master nodes, and can be used for worker
chapters/chapter04.rst:       the master nodes, or can run on separate nodes, but not on worker nodes).
chapters/chapter04.rst:     - An implementation must consist of at least one master node per availability zone or fault domain to ensure the
chapters/chapter04.rst:     - A master node must run at least the following Kubernetes control plane services: kube-apiserver, kube-scheduler
chapters/chapter04.rst:     - Workloads must not rely on the availability of the master nodes for the successful execution of their
chapters/chapter04.rst:       functionality (i.e. loss of the master nodes may affect non-functional behaviours such as healing and scaling,
chapters/chapter05.rst:-  **Kubernetes Master**: A master node in an unsecured boundary can lead to a
chapters/chapter05.rst:   potential threat to the running workloads. A master may be hardened in terms
chapters/chapter05.rst:-  Cluster management Network isolation can help protect the master node and
chapters/chapter05.rst:**Master node(s):**
```

**Review by CsatariGergely — APPROVED** [review]

**Review by tomkivlin — APPROVED** [review]

**Pull request merged**